tests/components/gree/test_climate.py

```diff
@@ -0,0 +1,781 @@
+"""Tests for gree component."""
+from datetime import timedelta
+
+from greeclimate.device import HorizontalSwing, VerticalSwing
+from greeclimate.exceptions import DeviceNotBoundError, DeviceTimeoutError
+import pytest
+
+from homeassistant.components.climate.const import (
+    ATTR_FAN_MODE,
+    ATTR_HVAC_MODE,
+    ATTR_PRESET_MODE,
+    ATTR_SWING_MODE,
+    DOMAIN,
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_COOL,
+    HVAC_MODE_DRY,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    PRESET_AWAY,
+    PRESET_BOOST,
+    PRESET_ECO,
+    PRESET_NONE,
+    PRESET_SLEEP,
+    SERVICE_SET_FAN_MODE,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_PRESET_MODE,
+    SERVICE_SET_SWING_MODE,
+    SERVICE_SET_TEMPERATURE,
+    SWING_BOTH,
+    SWING_HORIZONTAL,
+    SWING_OFF,
+    SWING_VERTICAL,
+)
+from homeassistant.components.gree.climate import (
+    FAN_MODES_REVERSE,
+    HVAC_MODES_REVERSE,
+    SUPPORTED_FEATURES,
+)
+from homeassistant.components.gree.const import (
+    DOMAIN as GREE_DOMAIN,
+    FAN_MEDIUM_HIGH,
+    FAN_MEDIUM_LOW,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_FRIENDLY_NAME,
+    ATTR_SUPPORTED_FEATURES,
+    ATTR_TEMPERATURE,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.setup import async_setup_component
+import homeassistant.util.dt as dt_util
+
+from .common import build_device_mock
+
+from tests.async_mock import DEFAULT as DEFAULT_MOCK, AsyncMock, patch
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+ENTITY_ID = f"{DOMAIN}.fake_device_1"
+
+
+@pytest.fixture
+def mock_now():
+    """Fixture for dtutil.now."""
+    return dt_util.utcnow()
+
+
+async def async_setup_gree(hass):
+    """Set up the gree platform."""
+    MockConfigEntry(domain=GREE_DOMAIN).add_to_hass(hass)
+    await async_setup_component(hass, GREE_DOMAIN, {GREE_DOMAIN: {"climate": {}}})
+    await hass.async_block_till_done()
+
+
+async def test_discovery_called_once(hass, discovery, device):
+    """Test discovery is only ever called once."""
+    await async_setup_gree(hass)
+    assert discovery.call_count == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)